src/Fetch_Abort.res

```diff
@@ -1,10 +1,39 @@
+module AbortEvent = {
+  type t
+}
+
 module Signal = {
   type t
 
+  /**
+  https://dom.spec.whatwg.org/#dom-abortsignal-any
+  this is new as of 2024
+   */
+  @scope("AbortSignal")
+  external any: array<t> => t = "any"
+
+  @scope("AbortSignal")
+  external timeout: int => t = "timeout"
+
+  @scope("AbortSignal")
+  external abort: 'reason => t = "abort"
+
   @get external aborted: t => bool = "aborted"
   @get @return(nullable) external reason: t => option<_> = "aborted"
 
   @send external throwIfAborted: t => unit = "throwIfAborted"
+
+  @set external onabort: (t, AbortEvent.t => unit) => unit = "onabort"
+
+  // inherited from EventTarget
+  // TODO: event options like once, passive, capture?
+  @send
+  external addEventListener: (t, @string [#abort(AbortEvent.t => unit)]) => unit =
+    "addEventListener"
```

Comment:
With uncurried mode we can make the options an optional argument, and have  all the fields of the options object optional. That would make it quite consistent with the JS API I think, and not too inconvenient to use. What do you think?

Reply:
Sounds good to me, I just wasn't sure how much API surface you'd wanted to cover here